doc/concepts/concepts.rst

```diff
@@ -1,13 +1,228 @@
+
+Planning Scene
+--------------
+
+.. image:: /_static/images/planning_scene.png
+
+The ``planning scene`` is used to represent the world around the robot and also stores the state of the robot itself. It is maintained by the ``planning scene monitor`` inside the ``move group`` node. The planning scene monitor listens to:
+
+- State Information: on the ``/joint_states`` topic
+
+- Sensor Information: using the world geometry monitor described below
+
+- World geometry information: from user input on the ``/planning_scene`` topic (as a planning scene diff).
+
+World Geometry Monitor
+----------------------
+
+The world geometry monitor builds world geometry using information from the sensors on the robot and from user input. It uses the ``occupancy map monitor`` described below to build a 3D representation of the environment around the robot and augments that with information on the ``planning_scene`` topic for adding object information.
+
+3D Perception
+-------------
+
+3D perception in MoveIt is handled by the ``occupancy map monitor``. The occupancy map monitor uses a plugin architecture to handle different kinds of sensor input as shown in the Figure above. In particular, MoveIt has inbuilt support for handling two kinds of inputs:
+
+- **Point clouds**: handled by the ``point cloud occupancy map updater`` plugin
+
+- **Depth images**: handled by the ``depth image occupancy map updater`` plugin
+
+Note that you can add your own types of updaters as a plugin to the occupancy map monitor.
+
+Octomap
+-------
+
+The Occupancy map monitor uses an `Octomap <http://octomap.github.io/>`_ to maintain the occupancy map of the environment. The **Octomap** can actually encode probabilistic information about individual cells although this information is not currently used in MoveIt. The **Octomap** can directly be passed into FCL, the collision checking library that MoveIt uses.
+
+Depth Image Occupancy Map Updater
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The depth image occupancy map updater includes its own *self-filter*, i.e. it will remove visible parts of the robot from the depth map. It uses current information about the robot (the robot state) to carry out this operation.
+
+
+**********
+Kinematics
+**********
+
+The Kinematics Plugin
+---------------------
+
+MoveIt uses a plugin infrastructure, especially targeted towards allowing users to write their own inverse kinematics algorithms. Forward kinematics and finding jacobians is integrated within the RobotState class itself. The default inverse kinematics plugin for MoveIt is configured using the [KDL](https://github.com/orocos/orocos_kinematics_dynamics) numerical jacobian-based solver. This plugin is automatically configured by the MoveIt Setup Assistant.
+
+IKFast Plugin
```

Comment:
IKFast hasn't been ported yet. Can you comment this out for now?

Reply:
Absolutely. Didn't realize that was the case.